File: src/main/java/org/bundleproject/discover/utils/FileUtils.java
package org.bundleproject.discover.utils;

import javax.imageio.ImageIO;
import java.awt.*;
import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.nio.file.CopyOption;
import java.nio.file.Files;

public class FileUtils {

    // https://stackoverflow.com/a/13379744/15301449
    public static void exportResource(String resourceName, File output, CopyOption... options) {
        try (InputStream is = FileUtils.class.getResourceAsStream(resourceName)) {
            if (is == null) {
                throw new IOException("Cannot get resource \"" + resourceName + "\"");
            }

            Files.copy(is, output.toPath(), options);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public static Image getResourceImage(String resourceName) {
        try (InputStream is = FileUtils.class.getResourceAsStream(resourceName)) {
            if (is == null) {
                throw new IOException("Cannot get resource \"" + resourceName + "\"");
            }

            return ImageIO.read(is);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static boolean deleteDirectory(File directoryToBeDeleted) {
        File[] allContents = directoryToBeDeleted.listFiles();
        if (allContents != null) {
            for (File file : allContents) {
                deleteDirectory(file);
            }
        }
        return directoryToBeDeleted.delete();
    }

    public static String removeFileExtension(String name) {
        return name.substring(0, name.lastIndexOf('.'));
    }

}

File: src/main/kotlin/org/bundleproject/discover/utils/FileUtils.kt
package org.bundleproject.discover.utils

import java.awt.Image
import java.io.File
import java.io.IOException
import java.nio.file.CopyOption
import java.nio.file.Files
import javax.imageio.ImageIO

object FileUtils {
    // https://stackoverflow.com/a/13379744/15301449
    fun exportResource(resourceName: String, output: File, vararg options: CopyOption?) {
        try {
            FileUtils::class.java.getResourceAsStream(resourceName).use { `is` ->
                if (`is` == null) {
                    throw IOException("Cannot get resource \"$resourceName\"")
                }
                Files.copy(`is`, output.toPath(), *options)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun getResourceImage(resourceName: String): Image? {
        try {
            FileUtils::class.java.getResourceAsStream(resourceName).use { `is` ->
                if (`is` == null) {
                    throw IOException("Cannot get resource \"$resourceName\"")
                }
                return ImageIO.read(`is`)
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return null
    }

    fun deleteDirectory(directoryToBeDeleted: File): Boolean {
        val allContents = directoryToBeDeleted.listFiles()
        if (allContents != null) {
            for (file in allContents) {
                deleteDirectory(file)
            }
        }
        return directoryToBeDeleted.delete()
    }

    fun removeFileExtension(name: String): String {
        return name.substring(0, name.lastIndexOf('.'))
    }
}
